common-libs/common/src/services/DidDocumentService/index.ts

```diff
@@ -1,28 +1,24 @@
 import KeysService from '../KeysService'
 import JoloDidDocument from './JoloDidDocument'
 import ElemDidDocument from './ElemDidDocument'
+import ElemAnchoredDidDocument from './ElemAnchoredDidDocument'
 import { parse } from 'did-resolver'
 import { LocalKeyVault } from './LocalKeyVault'
+import { DidResolver } from '../../shared/DidResolver'
 
 export { default as ElemDidDocument } from './ElemDidDocument'
 export { KeyVault } from './KeyVault'
 export { LocalKeyVault } from './LocalKeyVault'
 
 export default class DidDocumentService {
-  constructor(keysService: KeysService) {
+  constructor(keysService: KeysService, didResolver?: DidResolver) {
     const { didMethod } = keysService.decryptSeed()
 
-    let didDocumentService
-    switch (didMethod) {
-      case 'jolo':
-        didDocumentService = new JoloDidDocument(keysService)
-        break
-      case 'elem':
-        didDocumentService = new ElemDidDocument(new LocalKeyVault(keysService))
-        break
-    }
-
-    return didDocumentService
+    return {
+      jolo: new JoloDidDocument(keysService),
+      elem: new ElemDidDocument(new LocalKeyVault(keysService)),
+      'elem-anchored': new ElemAnchoredDidDocument(keysService, didResolver),
+    }[didMethod]
```

Comment:
Seems like a nice opportunity to also refactor this constructor into a factory (and mark constructor as deprecated, for removal in SDK v7).

Constructors shouldn't return values :)

Reply:
yes, you are right, I also thought about that. let's create a doc with possible refactorings and then when we have a time - we will work on that

Reply:
It is a really small refactoring and it makes sense within the context of this PR, so I'd suggest to do it here.

Reply:
ok